### PR TITLE
Form component

### PR DESCRIPTION
**Pull request by meeragk99**
im not sure why its not able to automatically merge.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,8 +7,10 @@
     "@testing-library/jest-dom": "^5.16.5",
     "@testing-library/react": "^13.4.0",
     "@testing-library/user-event": "^13.5.0",
+    "express-body-parser-error-handler": "^1.0.4",
```
Comment (FlyingRatMan) [inline package.json:13]: It doesn't seem that you are using express somewhere. Maybe you should remove it.

File: package.json [hunk 2]
```diff
@@ -7,8 +7,10 @@
...
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
+    "react-hook-form": "^7.42.1",
```
Comment (FlyingRatMan) [inline package.json]: Same goes here. It seems you are not using this hook form

File: package.json [hunk 3]
```diff
@@ -41,6 +43,9 @@
     ]
   },
   "devDependencies": {
-    "prettier": "^2.8.2"
+    "prettier": "^2.8.2",
+    "webpack": "^5.75.0",
```
Comment (FlyingRatMan) [inline package.json:51]: Same goes here

File: src/components/LocationForm.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
+import React, { useState } from "react";
+
+function LocationForm() {
+    const [title, setTitle] = useState("");
```
Comment (FlyingRatMan) [inline src/components/LocationForm.js]: You also need to get an Address from user

**Review by FlyingRatMan — CHANGES_REQUESTED** [review]
You should also provide functionality for this form. As we discussed on a meeting. You need to send a request with the address to a Google API and get coordinates. Then compile the whole information about location and send it to the backend.

File: package.json (hunk 1 above)
Comment (meeragk99) [reply package.json:13]: apparently, i do need them :)

File: public/index.html [hunk 5]
```diff
@@ -26,6 +26,9 @@
   <body>
     <noscript>You need to enable JavaScript to run this app.</noscript>
     <div id="root"></div>
+    <script src="https://unpkg.com/react@16/umd/react.development.js"></script>
```
Comment (FlyingRatMan) [inline public/index.html]: What are those scripts? Why do you need them?

File: package.json [hunk 6]
```diff
@@ -39,8 +39,5 @@
       "last 1 firefox version",
       "last 1 safari version"
     ]
-  },
-  "devDependencies": {
```
Comment (FlyingRatMan) [inline package.json:49]: We need this!

Comment (meeragk99) [reply package.json:49]: what do u mean? i dont think i removed anything other than u suggested XD

File: src/App.js [hunk 7]
```diff
@@ -1,3 +1,21 @@
+
+import LocationsList from "./components/LocationsList";
+import SearchBar from "./components/SearchBar";
+import LocationForm from "./components/LocationForm";
+
+function App() {
```
Comment (xchanti124) [inline src/App.js]: Why are you redefining and reexporting the app component which is also defined on line 22?

File: src/components/LocationForm.js [hunk 8]
```diff
@@ -0,0 +1,124 @@
+import React, { useState } from "react";
```
Comment (xchanti124) [inline src/components/LocationForm.js:1]: Prettier was not applied on this file. You can install an extension that's called prettier and then there should be an option to apply it on save in the settings (which would mean that prettier gets applied every time you save something).

File: src/App.js (hunk 7 above)
Comment (xchanti124) [reply src/App.js]: Also prettier is not applied here. Read the comment I left on the next file on how to apply it more naturally and easily.

**Review by xchanti124 — CHANGES_REQUESTED** [review]

File: src/components/LocationForm.js [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+        if (!imageLink) formErrors.push("imageLink is required");
+        return formErrors.length === 0;
+    };
+
+    const handleSubmit = async event => {
+        event.preventDefault();
+
+        if (validate()) {
+            console.log("Submitting form with:", { title, description, address,hashtags,category,imageLink});
+            let res = await fetch("http://localhost:3000/api/locations",{
```
Comment (xchanti124) [inline src/components/LocationForm.js]: You should probably use authenticatedFetch here (don't worry you'll get an explanation in a bit).

**Review by xchanti124 — CHANGES_REQUESTED** [review]

File: src/components/LocationForm.js [hunk 10]
```diff
@@ -25,7 +26,7 @@ function LocationForm() {
 
     if (validate()) {
       console.log("Submitting form with:", { title, description, address, hashtags, category, imageLink });
-      let res = await fetch("http://localhost:3000/api/locations", {
+      let res = authenticatedFetch("http://localhost:3000/api/locations", {
```
Comment (xchanti124) [inline src/components/LocationForm.js]: You accidentally deleted the `await` before `authenticatedFetch` . Other than that it looks very good!

**Review by xchanti124 — CHANGES_REQUESTED** [review]
Just a very small change, fix it and then you can merge I think!

File: src/App.js [hunk 11]
```diff
@@ -1,24 +1,28 @@
...
   <>
     <nav>
-      <NavLink to={"/new_location"}> CREATE NEW LOCATION </NavLink>
-      <NavLink to={"/"}> LOCATIONS </NavLink>
-      <NavLink to={"/login"}> LOGIN </NavLink>
+      <NavLink to={"/location_form"}>CREATE NEW LOCATION</NavLink>
+      <NavLink to={"/"}>LOCATIONS</NavLink>
+      <NavLink to={"/login"}>LOGIN</NavLink>
+      <NavLink to={"/about"}>
+        <button>ABOUT</button>
```
Comment (FlyingRatMan) [inline src/App.js]: You don't need `<button>` here

File: src/App.js [hunk 12]
```diff
@@ -1,24 +1,28 @@
...
-        <Route path={"details"} element={<Location />} />
-      </Route>
-      {/* <Route path={"/new_location"} element={<LocationForm/>} /> */}
-
+      <Route path={"/"} element={<LocationsList />} />
+      <Route path={"/details/:id"} element={<Location />} />
+      <Route path={"/location_form"} element={<LocationForm />} />
+      <Route path={"/about"} element={<About />} />
+      <Route path={"/login"} element={<div>Login page</div>} />
+      <Route path={"/404"} element={<Error404 />} />
```
Comment (FlyingRatMan) [inline src/App.js]: Why is it doubled? And the path should be "*". Or are we referencing this path somewhere?

**Review by FlyingRatMan — COMMENTED** [review]
For the future, I think you should do a separate branch for a different features. About pade shouldn't be on "form component" branch.

**Review by xchanti124 — APPROVED** [review]

**Pull request merged**